### DISCUSSION TITLE
Loving it

### DISCUSSION THREAD

**Discussion by laat0003** [gmdsi/GMDSI_notebooks#42] [General]
Hi,

Just wanted to say that the workflows demonstrated in these examples are absolute gold and do a great job of cementing the philosophy and theory published on the GMDSI website. Can't thank you guys enough.

Cheers,
Tariq

**Comment by rhugman** [comment]
> Hi,
> 
> Just wanted to say that the workflows demonstrated in these examples are absolute gold and do a great job of cementing the philosophy and theory published on the GMDSI website. Can't thank you guys enough.
> 
> Cheers, Tariq

Thanks Tariq, appreciated. Any input/feedback is also welcome. And keep an eye out for incoming changes!

**Reply by laat0003** [reply]
Will do thanks Rui.

As for feedback:
I really like the regression example as a teaching tool but if you over parametrize the example then you do end up with  larger errors in the prediction than if you under do it. This may cause some to think that too many parameters is worse than too little. A statement about the plausibility of overparameterizing a numerical groundwater model would help ward off those concerns. I know that you make a statement along these lines in subsequent tutorials.

PyEMU is my new beau! I'm not sure the value of this library, and how slick the examples are, will be evident to many people - except for those who regularly utilize much of the function in John's GW modelling utility suite plus PLPROC and OLPROC. If you could somehow convey the time saving aspects associated with the workflows presented, then I reckon you could turn this into a religion complete with public holidays....(Cue angel chorus)
The only way I can see to do that is to also step through a setup using all the different utilities which I guess is akin to car sales tactics. That is, first show the "get you from A to B model" then draw everyone's attention to to the Ferrari.

**Reply by rhugman** [reply]
Ah, but that's kind of the point :) Over-fitting doesn't mean good prediction. We will get into demonstrating this and how to avoid it in a groundwater modelling context in subsequent notebooks (just haven't gotten to it yet...watch this space!).

**Reply by mnfienen** [reply]
Thanks for replying @rhugman  - I meant to do so as well! Just want to second that comment that overfitting is still a real concern even though we now have the tools to cope with (and make good use of!) many parameters. Without care, bad predictions can be the result. That initial notebook is meant to show just how off the rails things can go without expert input (that's why we modelers are all still in the loop!).

**Reply by jtwhite79** [reply]
I wonder if the overfitting issue in that intro demo is because in this case the truth is actually simple (2 pars I think) but in practice the truth has nearly infinite parameters so that in the simple example you can actually add more parameters than there are parameters in "reality" (this only time I guess I can I say Ive seen "over-parameterization" haha) but in practice, even with the freyberg model,  you have to really work to get more parameters than the "truth"...The issue with overfitting to noise (whether it is just simple additive measurement noise, or structural noise) is still important to show tho.  Maybe some of this context would be nice in that notebook (maybe it is already there and I just havent looked...)

**Reply by laat0003** [reply]
Thanks all, @jtwhite79 is echoing my thoughts. The point that overfitting can make things much worse is demonstrated really well and as the primary "take home message", the example delivers. However, my concern is the, perhaps unintended, secondary message coming from the example suggesting that "less parameters are better than more" - because the error in the prediction remains small when using less parameters than "the truth" but it explodes when using more. 

Now consider this scenario, a modeller, who's only experience is practicing manual calibration with zones of piecewise constancy wants to walk the path of a Jedi and adopt the methods demonstrated in these tutorials. They already know that overfitting is bad because in the past, "expert" reviewers of their work raised this issue as "you are over fitting by using too many zones".  After consulting the oracle (Google) and obtaining the Bible (these tutorials), they do the regression example and naturally relate the polynomial to a numerical groundwater model. Upon completion of the tutorial they are now convinced that less zones is better for predictions and those "experts" were right all along.

Hence my suggestion for a qualifying statement about the plausibility of over-parametrisation of a numerical groundwater model versus the polynomial. Thanks again for taking the time to provide some discussion around this and apologies if I'm being a pain.

**Comment by laat0003** [comment]
Howdy,

Just went through the updated version of the regression example and it rocks. I especially like the closing section where you say "more parameters requires more care" but I would have gone with something more like this for the modelling nerds.

<img width="755" alt="Uncle_John" src="https://user-images.githubusercontent.com/49386249/229323216-232cb650-e916-4f26-8174-493bd9d7f92d.png">
